hedera-node/src/main/java/com/hedera/services/txns/file/FileAppendTransitionLogic.java

```diff
@@ -58,43 +58,24 @@ public FileAppendTransitionLogic(HederaFs hfs, TransactionContext txnCtx) {
 
 	@Override
 	public void doStateTransition() {
-		var op = txnCtx.accessor().getTxn().getFileAppend();
+		/* --- Extract from gRPC --- */
+		final var op = txnCtx.accessor().getTxn().getFileAppend();
+		final var target = op.getFileID();
+		final var data = op.getContents().toByteArray();
 
-		try {
-			var target = op.getFileID();
-			var data = op.getContents().toByteArray();
+		/* --- Perform validations --- */
+		final Optional<HFileMeta> attr = hfs.exists(target) ? Optional.of(hfs.getattr(target)) : Optional.empty();
+		classify(attr);
```

Comment:
nit: Instead of having another `classify` method that is kind of ambiguous what it actually does, I think that it would be better to have the validations in the `doStateTransition` directly as the other transition operations